File: src/algorithms/sort/insertion/java/InsertionSort.java
package algorithms.sort.insertion.java;

public class InsertionSort {

    public static void sort(int[] arr){
        // Starting with i = 1 because first element is sorted
        for(int i = 1; i < arr.length; i++ ){
            int currentValue = arr[i];

            int j = i -1;

            while(j >= 0 && arr[j] > currentValue){
                arr[j +1] = arr[j];
                j--;
            }

            arr[j + 1] = currentValue;
        }
    }

}

File: src/algorithms/sort/insertion/js/insertionSort.js

/**
 * 
 * @param {Array} arr 
 */
function insertionSort(arr){

    for(let i = 1; i < arr.length; i++){
        let currentValue = arr[i]; 

        let j = i -1; 

        while(j >= 0 && arr[j] > currentValue){
            arr[j +1] = arr[j]; 
            j--
        }

        arr[j +1] = currentValue


    }

}


const array = [1,5,3,2,202,24,26,33]

insertionSort(array); 

console.log(array);
